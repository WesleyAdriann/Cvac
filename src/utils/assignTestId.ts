import { Platform } from 'react-native'

interface ITestID {
  accessible?: boolean
  testID: string
  accessibilityLabel?: string
}

type TComponentType =
  | 'View'
  | 'ScrollView'
  | 'TextInput'
  | 'Text'
  | 'SafeAreaView'
  | 'Pressable'
  | 'Modal'

export const assignTestId = (
  componentType: TComponentType,
  componentId: string
): ITestID => {
  const id = `${componentType}_${componentId}`
  return Platform.OS === 'android'
    ? {
        testID: id,
        accessible: true,
        accessibilityLabel: id
      }
    : {
        testID: id,
        accessible: false
      }
}
export const mockTestID = (
  componentType: TComponentType,
  componentId: string
): ITestID => assignTestId(componentType, componentId)
