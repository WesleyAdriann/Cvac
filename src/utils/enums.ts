export enum ECalendarsName {
  'kid' = 'Crian√ßa',
  'elder' = 'Idoso',
  'adult' = 'Adulto',
  'teen' = 'Adolescente'
}
