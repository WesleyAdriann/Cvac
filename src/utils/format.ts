export const formatDate = (value: string) => {
  const clearDate = value.replace(/\D/g, '')

  if (clearDate.length === 8) return value

  const day = clearDate.slice(0, 2)
  const month = clearDate.slice(2, 4)
  const year = clearDate.slice(4, 8)

  if (!month) return day
  if (!year) return `${day}/${month}`
  return `${day}/${month}/${year}`
}
