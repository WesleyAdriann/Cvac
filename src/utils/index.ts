export * from './assignTestId'
export * from './withProviders'
export * from './format'
export * from './logs'
