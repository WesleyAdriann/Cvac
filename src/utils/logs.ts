export const log = (TAG: string, ...args: any[]) => {
  if (__DEV__) console.log(`[${TAG}] `, ...args)
}
