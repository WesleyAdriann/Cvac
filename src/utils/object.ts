export const isEmpty = (obj: object | null) => {
  if (!obj) return true
  // eslint-disable-next-line no-unreachable-loop
  for (const key in obj) return false
  return true
}
