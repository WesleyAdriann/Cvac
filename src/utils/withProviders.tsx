import React from 'react'
import { Provider as PaperProvider } from 'react-native-paper'
import { ThemeProvider } from 'styled-components'

import { theme, paperTheme } from '../tokens'

export const TestsProviders: React.FC = ({ children }) => (
  <PaperProvider theme={paperTheme} >
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  </PaperProvider>
)
