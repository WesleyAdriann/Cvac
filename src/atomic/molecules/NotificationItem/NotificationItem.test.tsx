import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react-native'

import { AllTheProviders as wrapper, mockTestID } from '~/utils'

import { NotificationItem, INotificationItem } from './index'

describe('Molecules/NotificationItem', () => {
  const props: INotificationItem = {
    testID: 'NotificationItem',
    onPress: jest.fn(),
    day: '25',
    month: 'Janeiro',
    descriptions: [],
    title: 'Vaccina',
    year: '2022'
  }

  const elContainer = mockTestID('Pressable', props.testID!).testID

  const renderComponent = () => render(<NotificationItem {...props} />, { wrapper })
  let component = renderComponent()

  beforeEach(() => {
    jest.clearAllMocks()
    cleanup()
    component = renderComponent()
  })

  it(`should render component #${elContainer}`, async () => {
    const sut = await component.findByTestId(elContainer)
    expect(sut).toBeTruthy()
  })

  it(`should call onPress prop when press #${elContainer}`, async () => {
    const sut = await component.findByTestId(elContainer)
    expect(props.onPress).not.toBeCalled()
    fireEvent(sut, 'press')
    expect(props.onPress).toBeCalled()
  })

  it('should render snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })
})
