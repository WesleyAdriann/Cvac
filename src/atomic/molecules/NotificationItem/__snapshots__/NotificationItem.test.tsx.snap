// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/NotificationItem should render snapshot 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={false}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="Pressable_NotificationItem"
  >
    <View
      accessible={false}
      flexDirection="row"
      style={
        Array [
          Object {
            "flexDirection": "row",
            "marginBottom": 0,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 0,
            "paddingBottom": 0,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 0,
          },
          Object {
            "borderColor": "#0000001f",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 1,
            "marginBottom": 8,
          },
        ]
      }
      testID="View_Flex"
    >
      <View
        accessible={false}
        alignItems="center"
        flex={1.2}
        justifyContent="center"
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexDirection": "column",
              "flexGrow": 1.2,
              "flexShrink": 1,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
            Object {
              "backgroundColor": "#F1FAFC",
            },
          ]
        }
        testID="View_Flex"
      >
        <Text
          accessible={false}
          size={24}
          style={
            Array [
              Object {
                "color": "#424242",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "color": "#424242",
                  "fontSize": 24,
                  "fontWeight": "bold",
                  "marginBottom": 0,
                  "textAlign": "left",
                  "textDecorationColor": "black",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                },
                Object {
                  "textTransform": "capitalize",
                },
              ],
            ]
          }
          testID="Text_Text"
          weight="bold"
        >
          25
        </Text>
        <Text
          accessible={false}
          style={
            Array [
              Object {
                "color": "#424242",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "color": "#424242",
                  "fontSize": 16,
                  "fontWeight": "normal",
                  "marginBottom": 0,
                  "textAlign": "left",
                  "textDecorationColor": "black",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                },
                Object {
                  "textTransform": "capitalize",
                },
              ],
            ]
          }
          testID="Text_Text"
        >
          Janeiro
        </Text>
        <Text
          accessible={false}
          style={
            Array [
              Object {
                "color": "#424242",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "color": "#424242",
                  "fontSize": 16,
                  "fontWeight": "normal",
                  "marginBottom": 0,
                  "textAlign": "left",
                  "textDecorationColor": "black",
                  "textDecorationLine": "none",
                  "textDecorationStyle": "solid",
                },
                Object {
                  "textTransform": "capitalize",
                },
              ],
            ]
          }
          testID="Text_Text"
        >
          2022
        </Text>
      </View>
      <View
        accessible={false}
        flex={3}
        justifyContent="center"
        paddingStyle={24}
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexDirection": "column",
              "flexGrow": 3,
              "flexShrink": 1,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "paddingBottom": 24,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 24,
            },
          ]
        }
        testID="View_Flex"
      >
        <View
          accessible={false}
          marginStyle="0 8px 8px 0"
          style={
            Array [
              Object {
                "flexDirection": "column",
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 8,
                "marginTop": 0,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
              },
            ]
          }
          testID="View_Flex"
        >
          <Text
            accessible={false}
            size={22}
            style={
              Array [
                Object {
                  "color": "#424242",
                  "fontFamily": "System",
                  "fontWeight": "400",
                },
                Object {
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "color": "#424242",
                    "fontSize": 22,
                    "fontWeight": "normal",
                    "marginBottom": 0,
                    "textAlign": "left",
                    "textDecorationColor": "black",
                    "textDecorationLine": "none",
                    "textDecorationStyle": "solid",
                  },
                ],
              ]
            }
            testID="Text_Text"
          >
            Vaccina
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
